lib/logstash/plugin_mixins/event_support.rb

```diff
@@ -0,0 +1,79 @@
+require 'logstash/namespace'
+require 'logstash/event'
+
+require 'logstash/plugin_mixins/event_support/event_factory'
+require 'logstash/plugin_mixins/event_support/event_target_decorator'
+
+module LogStash
+  module PluginMixins
+    module EventSupport
+
+      def event_factory
+        @_event_factory ||=
+          if ecs_compatibility?
+            EventFactory::INSTANCE
+          else
+            LegacyEventFactory::INSTANCE
+          end
+      end
+
+      def event_factory=(factory)
+        @_event_factory = factory
+      end
+
+      # Creates a new event using the set event factory.
+      # @param target_namespace an optional namespace (prefix) for data in event
+      # @param event_factory
+      # @return [LogStash::Event]
+      def new_event(data = {}, target_namespace: nil, event_factory: self.event_factory)
```

Comment:
a generar theme with `target: namespace` is to keep internals (`@refs`) as is 
and **only namespace "plain-old" data**